jlm/llvm/backend/RvsdgToIpGraphConverter.cpp

```diff
@@ -277,25 +278,58 @@ RvsdgToIpGraphConverter::ConvertEmptyGammaNode(const rvsdg::GammaNode & gammaNod
       continue;
     }
 
-    const auto node = rvsdg::output::GetNode(*predicate);
-    if (is<rvsdg::match_op>(node))
+    const auto matchNode = rvsdg::output::GetNode(*gammaPredicate);
+    if (is<rvsdg::match_op>(matchNode))
     {
-      const auto matchOperation = util::AssertedCast<const rvsdg::match_op>(&node->GetOperation());
-      const auto defaultAlternative = matchOperation->default_alternative();
-      const auto condition = Context_->GetVariable(node->input(0)->origin());
-      const auto trueAlternative =
-          defaultAlternative == 0 ? Context_->GetVariable(output1) : Context_->GetVariable(output0);
-      const auto falseAlternative =
-          defaultAlternative == 0 ? Context_->GetVariable(output0) : Context_->GetVariable(output1);
+      const auto matchOperation =
+          util::AssertedCast<const rvsdg::match_op>(&matchNode->GetOperation());
+      assert(matchOperation->nalternatives() == 2);
+
+      const auto matchOrigin = Context_->GetVariable(matchNode->input(0)->origin());
+      const auto caseValue = matchOperation->begin()->first;
+      const auto caseSubregion = matchOperation->begin()->second;
+      const auto defaultSubregion = matchOperation->default_alternative();
+      const auto numMatchBits = matchOperation->nbits();
+      assert(caseSubregion != defaultSubregion);
+
+      const variable * selectPredicate = nullptr;
+      const variable * trueAlternative = nullptr;
+      const variable * falseAlternative = nullptr;
+      if (numMatchBits == 1 && caseValue == caseSubregion)
+      {
+        // We have an identity mapping:
+        // 1. 0 -> 0, default 1, or
+        // 2. 1 -> 1, default 0
+        // There is no need to insert operations for the select predicate
+        selectPredicate = matchOrigin;
+        trueAlternative = Context_->GetVariable(output1);
+        falseAlternative = Context_->GetVariable(output0);
+      }
+      else
+      {
+        // FIXME: This will recreate the select predicate operations for each gamma output for
+        // which we create a select.
```

Comment:
Please ignore this for now. My plan is to move the entire select creation out of the RvsdgToIpGraph convert into its own transformation pass, and invoke as one of the last passes in the optimization pipeline. The entire IfConversion story of gamma nodes does not really belong in the RvsdgToIpGraph converter.

The FIXME will be handled once this is done.